corehq/motech/dhis2/forms.py

```diff
@@ -1,20 +1,367 @@
+    from corehq.motech.views import ConnectionSettingsListView
+
+    connection_settings = ConnectionSettings.objects.filter(domain=domain)
+    url = reverse(ConnectionSettingsListView.urlname,
+                  kwargs={'domain': domain})
+    return forms.ModelChoiceField(
+        label=_("Connection Settings"),
+        queryset=connection_settings,
+        required=True,
+        help_text=_(f'<a href="{url}">Add/Edit Connections Settings</a>')
+    )
+
+
+def get_ucr_field(domain):
+    from corehq.apps.userreports.views import UserConfigReportsHomeView
+
+    ucrs = get_report_configs_for_domain(domain)
+    url = reverse(UserConfigReportsHomeView.urlname,
+                  kwargs={'domain': domain})
+    return forms.ChoiceField(
+        label=_("User Configurable Report"),
+        choices=[(ucr.get_id, ucr.title) for ucr in ucrs],
+        required=True,
+        help_text=_(
+            'DataSet Maps map CommCare UCRs to DHIS2 DataSets. '
+            '<a href="https://github.com/dimagi/commcare-hq/blob/master/corehq/motech/dhis2/README.md#datasets" '
+            f'target="_new">More info</a>. '
+            f'Go to <a href="{url}">Configurable Reports</a> to define a UCR.')
+    )
+
+
+class DataValueMapBaseForm(forms.ModelForm):
+    column = forms.CharField(
+        label=_('Column'),
+        required=True,
+    )
+    data_element_id = forms.CharField(
+        label=_('Data element ID'),
+        required=True,
+        validators=[RegexValidator(DHIS2_UID_RE, DHIS2_UID_MESSAGE)],
+    )
+    category_option_combo_id = forms.CharField(
+        label=_('Category option combo ID'),
+        required=False,
+        validators=[RegexValidator(DHIS2_UID_RE, DHIS2_UID_MESSAGE)],
+    )
+    comment = forms.CharField(
+        label=_('Comment'),
+        required=False,
+    )
+
+    class Meta:
+        model = SQLDataValueMap
+        fields = [
+            'column',
+            'data_element_id',
+            'category_option_combo_id',
+            'comment',
+        ]
+
+    def __init__(self, dataset_map, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.dataset_map = dataset_map
+        self.helper = self.get_form_helper()
+
+    def get_form_helper(self):
+        raise NotImplementedError
+
+    def save(self, commit=True):
+        self.instance.dataset_map = self.dataset_map
+        return super().save(commit)
+
+
+class DataValueMapCreateForm(DataValueMapBaseForm):
+
+    def get_form_helper(self):
+        helper = FormHelper()
+        helper.form_style = 'inline'
+        helper.form_show_labels = False
+        helper.layout = crispy.Layout(
+            twbscrispy.InlineField('column'),
+            twbscrispy.InlineField('data_element_id'),
+            twbscrispy.InlineField('category_option_combo_id'),
+            twbscrispy.InlineField('comment'),
+            twbscrispy.StrictButton(
+                mark_safe(f'<i class="fa fa-plus"></i> {_("Add")}'),
```

Comment:
Can you add a comment to hide this (safe) usage of `mark_safe` from automated checks?

```suggestion
                mark_safe(f'<i class="fa fa-plus"></i> {_("Add")}'),  # nosec: no user input
```